worker/caaWorker/caaWorker.go

```diff
@@ -0,0 +1,164 @@
+package caaWorker
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/miekg/dns"
+	"github.com/mozilla/tls-observatory/worker"
+)
+
+var (
+	workerName = "caaWorker"
+	workerDesc = "Checks domains DNS records for a CAA record and reports it."
+)
+
+func init() {
+	worker.RegisterWorker(workerName, worker.Info{Runner: new(eval), Description: workerDesc})
+}
+
+// Result describes the result produced by CAAWorker
+type Result struct {
+	HasCAA       bool     `json:"has_caa"`
+	Host         string   `json:"host"`
+	IssueCAs     []string `json:"issue_cas"`
+	IssueWildCAs []string `json:"issuewild_cas"`
+}
+
+type eval struct{}
+
+// Run implements the worker interface.It is called to get the worker results.
+func (e eval) Run(in worker.Input, resChan chan worker.Result) {
+	result := worker.Result{WorkerName: workerName, Success: true}
+	caaRes := Result{}
+
+	hostPieces := strings.Split(in.Target, ".")
+	for i := 0; i < len(hostPieces); i++ {
+		host := strings.Join(hostPieces[i:], ".")
+
+		msg := new(dns.Msg)
+		msg.SetQuestion(dns.Fqdn(host), dns.TypeCAA)
+
+		client := dns.Client{}
+		res, _, err := client.Exchange(msg, "8.8.8.8:53")
```

Comment:
Should this use the locally configured resolver?

Reply:
I am not sure about what the caching behaviour is and if CAA records and fetched correctly.

Reply:
But people running tlsobs locally may not permit outbound DNS to Google, in which case this worker will fail. I think reading the resolver from /etc/resolv.conf and falling back to 8.8.8.8 is the right thing to do.

Reply:
Done, nice catch :+1: